cmd/stackdriver-prometheus-sidecar/main.go

```diff
@@ -598,12 +609,24 @@ func (s *stackdriverClientFactory) Name() string {
 	return s.url.String()
 }
 
+// fileClientFactory allows creating a StorageClient which writes to
+// a newly created file under dir.
 type fileClientFactory struct {
+	dir    string
 	logger log.Logger
 }
 
-func (f *fileClientFactory) New() stackdriver.StorageClient {
-	return file.NewFileClient(f.logger)
+// New Creates a new file for each StorageClient, so when Multiple
+// StorageClients execute Store(), these clients write to their own
+// files without race condition.
```

Comment:
Rephrase and document the type instead, "this class isn't thread-safe".

Reply:
Can you clarify more what you meant here? I think the `fileClientFactory` itself is thread-safe because there is no race condition between each client it creates. Even 2 instances of `fileClientFactory` configure the same path, there created client should not have race condition as well.

Reply:
Oh, I see what you meant. I still don't think the factory should document how the clients will use the created class. I think you should document the class API, such as "each instance will write to a different file" and "the returned class isn't thread-safe".

Reply:
Updated the comment.